ckanext/short_urls/blueprints/short_urls_blueprint.py

```diff
@@ -1,41 +1,43 @@
 from flask import Blueprint
+from flask.views import MethodView
 import ckan.plugins.toolkit as toolkit
 from ckan.lib import helpers as h
 from ckanext.short_urls.model import ObjectType
-from ckanext.short_urls.logic import short_url_get
+from ckanext.short_urls.logic import get_short_url_from_code
 
 
 short_urls_blueprint = Blueprint(
-    u'short_urls_blueprint',
+    u'short_urls',
     __name__
 )
 
 
-def short_url_redirect(short_url_code):
-    short_url = short_url_get(short_url_code)
-    object_type = short_url['object_type']
-    if short_url['object_state'] == 'active':
+class ShortUrlRedirect(MethodView):
```

Comment:
MethodView is a "template renderer" flask metaclass. You're not really generating views here.
I suggest you go with simple blueprint approach with registering a single function as a callback. Take a look here:
ckanext.unaids.blueprints.unaids_dataset_transfer.process_dataset_transfer

You can even go more modern and use `@annotations`